### ISSUE TITLE
Header title covers half of back button on iOS

### ISSUE THREAD

**Issue by davidskaarup** [react-community/react-navigation#197]
I think a recent commit caused this, it used to work fine. Now you can only pop the scene by pressing the left half of the back button.

<img width="487" alt="screen shot 2017-02-06 at 13 56 09" src="https://cloud.githubusercontent.com/assets/9447649/22647972/76ce9572-ec74-11e6-8591-9e471774fa09.png">

**Comment by satya164** [comment]
cc @grabbou

**Comment by michaelduminy** [comment]
I think there may be an issue with the recent header changes on Android.

We're using master at https://github.com/react-community/react-navigation/commit/eece5ad56a074ef2046c36f43ccb3ef66f5cc312. When we go to current this happens:

Main screen before:
![screen shot 2017-02-06 at 5 13 15 pm](https://cloud.githubusercontent.com/assets/4092035/22653022/07c6bdae-ec91-11e6-9312-993bede699e5.png)

Main screen after:
![screen shot 2017-02-06 at 5 11 20 pm](https://cloud.githubusercontent.com/assets/4092035/22653043/12218dce-ec91-11e6-985f-13083b18539b.png)

The title is set to `left: 10`.

Also in our modal screens the header used to look like this:
![screen shot 2017-02-06 at 5 13 22 pm](https://cloud.githubusercontent.com/assets/4092035/22653113/3c3f256c-ec91-11e6-9b5a-d5e2b91a14ef.png)

Now they look like this:
![screen shot 2017-02-06 at 5 11 41 pm](https://cloud.githubusercontent.com/assets/4092035/22653124/41b39226-ec91-11e6-8ff9-4774f1b7051e.png)

Our custom back button is not rendered or clickable.

**Comment by satya164** [comment]
Sorry for the trouble guys. Can you try reverting https://github.com/react-community/react-navigation/commit/677625ef6a4253c5cdd3a02a2b0293eb640358c3 and see if it still happens?

**Comment by davidskaarup** [comment]
Reverting that commit seems to solve the problem. I forgot to mention that the back button label had also disappeared, as can be seen on the screenshot. That also works again after reverting. It would be nice to have an option to hide the back button label and only show the arrow, but that should probably have its own issue.

**Comment by ericvicenti** [comment]
If you add an example with a similar header, we can make sure to test it before landing any more changes to the header.

Usually we would add a test case to prevent this sort of thing, but this is a visual bug and the best way to avoid future breakage is to add this use case to the examples.

**Comment by grabbou** [comment]
Sorry for the troubles. I'll take a look into these issues now. Some regressions are possible as the layout changed a lot, but they are minors. Thanks for testing it out!

**Comment by grabbou** [comment]
Reverted original PR. I will work on improving that and re-submit tomorrow with better layout.

**Comment by mmazzarolo** [comment]
> Reverted original PR.
  
Just pulled from the `master` branch, I'm still having the issue 🦆  
Am i missing something?

**Comment by mkonicek** [comment]
> Reverted original PR. 

> Just pulled from the master branch, I'm still having the issue :duck:

@grabbou Is it fixed now? Should the issue be closed?

**Comment by grabbou** [comment]
@mmazzarolo which one? There are two glitches described here. There's no way my work affects you by any chance as the changes are reverted. It is possible that you have met another edge-case though, so it would be great to tackle it as well.

**Comment by mmazzarolo** [comment]
@grabbou sorry, I was talking about the first one :)
I'll give you a feedback tomorrow, I can't test it right now

**Comment by michaelduminy** [comment]
@grabbou sorry I hijacked this issue with my header problems. I'm currently on https://github.com/react-community/react-navigation/commit/c35a7065b862570aeb342dda5cf61190f5cdfb79, the revert commit, and the problem is fixed.

**Comment by grabbou** [comment]
Yeah, makes a lot of sense.

@mmazzarolo can you check if latest version works for you?

**Comment by mmazzarolo** [comment]
@grabbou mhhh that's REALLY weird: I'm starting to think that the issue I'm having is not related to this one.  
Basically: 
1. I have a main screen with a custom right button that instantly disappear (after something like 50ms) after the screen loads.
2. This custom right button is not visible but it is still touchable. Pressing it pushes another route in stack and from now on every single route has the issue of OP.  

The app works perfectly if I don't press that magic button (even when pushing/resetting the stack).  

Starting debugging it right now.

**Comment by satya164** [comment]
@mmazzarolo this seems to be related to native animations. can you disable native animations and try? Or maybe see if https://github.com/facebook/react-native/pull/11819 fixes your issue?

**Comment by mmazzarolo** [comment]
@satya164 I just tried @grabbou 's test with the background colors and it seems that the issue is the title style:
```javascript
  static navigationOptions = {
    header: ({ navigate }) => ({
      style: { backgroundColor: 'green' },
      titleStyle: { backgroundColor: 'red' },
      title: strings.SCREEN_TITLE_NEW_APPLIANCE,
    }),
  };
```


![simulator screen shot 09 feb 2017 11 45 08](https://cloud.githubusercontent.com/assets/9536354/22780041/77f7cc48-eebd-11e6-855e-40aba2d51e40.png)
![simulator screen shot 09 feb 2017 11 45 12](https://cloud.githubusercontent.com/assets/9536354/22780042/77f96d0a-eebd-11e6-84c7-0b4eb6a3d56f.png)

I'll try to reset everything one last time, I'm starting to think that the repo version I'm downloading is still an older one.

**Comment by satya164** [comment]
@mmazzarolo Thanks for looking into it. Can you send a PR updating the horizontal margin on iOS here? https://github.com/react-community/react-navigation/blob/master/src/views/HeaderTitle.js

**Comment by mmazzarolo** [comment]
What a noobie mistake, it was a yarn caching issue (`watchman watch-del-all && rm -rf node_modules/ && npm cache clean && npm prune && yarn cache clean` was not enough).  
It is working fine now (well, actually I can't see the text on the back button anymore but that's another kind of beast).  
  
Thank you for the support and sorry for the mistake, I think that the issue can be closed now.

**Comment by grabbou** [comment]
>  I can't see the text on the back button anymore

I have disabled it temporarily because it caused some regressions on Android. I am working on fixing it today.

I am going to close it as I am pretty sure there's no way there are any glitches here (spent quite a lot to fix them yesterday). Thanks @mmazzarolo for staying with us and looking for a possible solution!

**Comment by tdurand** [comment]
I'm not sure if this is supposed to be fixed or if the fix is in progress, but I'm running in this issue as well, I just pulled master, here is a Screenshot:

![screen shot 2017-02-13 at 17 15 19](https://cloud.githubusercontent.com/assets/533590/22891756/9f897ada-f210-11e6-8eef-5e8d62f941bc.png)

**Comment by tdurand** [comment]
Well, after resetting the yarn cache and really pulling the new version, seems it's kind of fixed, but still overlapping the text part of the back button.

![screen shot 2017-02-13 at 17 28 02](https://cloud.githubusercontent.com/assets/533590/22892156/e3d3a0f2-f211-11e6-99b1-ec83cb3ea201.png)

Thanks you guys for the work on that library, much needed in the React Native ecosystem !

**Comment by tdurand** [comment]
Forget what I said, inspector says it overlaps, but the touch is working correctly @grabbou 👍 👍

**Comment by satya164** [comment]
Note for people using yarn: Yarn doesn't cache GIT repo dependencies properly. Instead of caching by commit hash it caches by the version number specified in `package.json`. So if the version number hasn't changed in the repo, it won't install the latest version, and will just install the cached version. Also if you install from npm and it has cached git repo with same version, it'll install the git version instead. So be careful.

**Comment by tdurand** [comment]
Good catch @satya164 , seems it will be fixed with the 0.20 release: https://github.com/yarnpkg/yarn/issues/1573#issuecomment-275149958

**Comment by mmazzarolo** [comment]
thank you @satya164 💘
